nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/AbstractMarkLogicProcessor.java

```diff
@@ -203,8 +205,9 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
 
     protected ServerTransform buildServerTransform(ProcessContext context) {
         ServerTransform serverTransform = null;
-        Objects.requireNonNull(context.getProperty(TRANSFORM), "TRANSFORM property should not be null");
-        final String transform = context.getProperty(TRANSFORM).getValue();
+        PropertyValue transformProperty = context.getProperty(TRANSFORM);
```

Comment:
Not needed now, but this pattern is repeated so often that it'd be worth making helper methods in the parent class to provide syntactic sugar so you can just do e.g.:

```
String value = getPropertyValue(context, TRANSFORM);
```

And then parent method would do the null check. Can add variants that evaluate based on the context or the FlowFile too. Just gets it all back down to one line of code per property value.